src/config.js

```diff
@@ -14,40 +14,62 @@ type ApplyConfigToArgvParams = {|
   argv: Object,
   configObject: Object,
   defaultValues: Object,
+  subCommandDefaultValues: Object,
   configFileName: string,
+  commandExecuted?: string,
 |};
 
 export function applyConfigToArgv({
   argv,
   configObject,
   defaultValues = {},
+  subCommandDefaultValues,
   configFileName,
+  commandExecuted,
 }: ApplyConfigToArgvParams): Object {
-  const newArgv = {...argv};
+  let newArgv = {...argv};
   for (const option in configObject) {
+
+    validateOption(option);
+    if (option === commandExecuted) {
+      newArgv = applyConfigToArgv({
+        argv,
+        configObject: configObject[option],
+        defaultValues: subCommandDefaultValues,
+        subCommandDefaultValues,
```

Comment:
@kumar303, here I am being forced to pass `subCommandDefaultValues` again because if I make it optional, Flow starts complaining that it can be undefined on the previous line. Is there a way to get around this?